ISISNeutronMuon/MDANSE
# Improved output file names

**Description of work**
Introduces sensible suggestions of output file names. Prevents from writing into a file that it already open (which would fail anyway).

This PR does **NOT** prevent overwriting existing files, since users could have valid reasons to do so. The initial filename suggested by the GUI will prevent overwriting, and the file dialog will issue a warning when overwriting. Once a job has started, the output filename becomes protected, preventing from running another job with the same output filename. However, after the job has ended, it will be possible to overwrite the file ONLY after removing the output file from the GUI (e.g. from the plot creator of from the trajectory viewer).

closes #599 
closes #608 

**Fixes**
- improved communication between widgets and session settings,
- created a list of protected filenames in the session object,
- added filename suggestions to the output file widgets.

**To test**
Please run at least two trajectory converters. See if the output trajectory widget suggests a valid path and filename. It should be impossible to run the same analysis twice without changing the output filename.
Try running an analysis. Check if the suggested output filename is valid.
Load an analysis result into the plotter. Check if you can run an analysis that overwrites the file. It should be impossible to overwrite as long as the file is on the list of open files in the plot creator.
